metadata_service/proxy/neo4j_proxy.py

```diff
@@ -682,6 +683,35 @@ def get_popular_tables(self, *, num_entries: int =10) -> List[PopularTable]:
             popular_tables.append(popular_table)
         return popular_tables
 
+    @timer_with_counter
+    def get_user_detail(self, *, user_id: str) -> Union[UserEntity, None]:
+        """
+        Retrieve user detail based on user_id(email).
+
+        :param user_id: the email for the given user
+        :return:
+        """
+
+        query = textwrap.dedent("""
+        MATCH (user:User {key: $user_id}) RETURN user as user_record
+        """)
+
+        record = self._execute_cypher_query(statement=query,
+                                            param_dict={'user_id': user_id})
+        result = None
+        if record:
+            record = record.single()['user_record']
+            result = UserEntity(email=record['email'],
+                                first_name=record.get('first_name'),
+                                last_name=record.get('last_name'),
+                                full_name=record.get('full_name'),
+                                is_active=record.get('is_active'),
+                                github_username=record.get('github_username'),
+                                team_name=record.get('team_name'),
+                                slack_id=record.get('slack_id'),
+                                employee_type=record.get('employee_type'))
+        return result
```

Comment:
I think we can throw `NotFoundException` if not exist instead of returning `None`

Reply:
updated